processors/tenantidprocessor/tenantidprocessor.go

```diff
@@ -21,6 +21,33 @@ type processor struct {
 
 var _ processorhelper.TProcessor = (*processor)(nil)
 
+var _ processorhelper.MProcessor = (*processor)(nil)
+
+func (p *processor) ProcessMetrics(ctx context.Context, metrics pdata.Metrics) (pdata.Metrics, error) {
+	md, ok := metadata.FromIncomingContext(ctx)
+	if !ok {
+		p.logger.Error("Could not extract headers from context", zap.Int("num-metrics", metrics.MetricCount()))
```

Comment:
I don't think we should be returning the error aswell as logging it. We probably should just return it.

Reply:
Okay. I added similar to https://github.com/hypertrace/hypertrace-collector/blob/main/processors/tenantidprocessor/tenantidprocessor.go#L28, I will update at both places